File: back-end/src/main/java/dk/eegelund/bikerides/core/CategoryService.java
package dk.eegelund.bikerides.core;


import dk.eegelund.bikerides.model.CategoryDto;

import java.util.List;

public interface CategoryService {
    CategoryDto createCategory(CategoryDto category);
    void deleteCategory(Long id);
    List<CategoryDto> getCategories();
    CategoryDto updateCategory(Long id, CategoryDto category);
}

File: front-end/src/app/service/category.service.ts
import {Observable} from "rxjs";
import {HttpClient} from "@angular/common/http";
import {Injectable} from "@angular/core";
import {Category} from "../model/category.model";

@Injectable()
export class CategoryService {
  private baseUrl = '/api/v1/categories';

  constructor(private http: HttpClient) { }

  getCategoryList(): Observable<Category[]> {
    return this.http.get<Category[]>(`${this.baseUrl}`);
  }

  getCategoryById(id: number): Observable<Category> {
    return this.http.get<Category>(`${this.baseUrl}/${id}`);
  }

  createCategory(category: Category): Observable<Category> {
    return this.http.post<Category>(`${this.baseUrl}`, category);
  }

  updateCategory(id: number, category: Category): Observable<Category> {
    return this.http.put<Category>(`${this.baseUrl}/${id}`, category);
  }
}
